src/laptop_battery_monitor/scripts/DockDriveActionClient.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+import roslib; roslib.load_manifest('kobuki_auto_docking')
+import rospy
+
+import actionlib
+from kobuki_msgs.msg import AutoDockingAction, AutoDockingGoal
+from actionlib_msgs.msg import GoalStatus
+from diagnostic_msgs.msg import DiagnosticArray
+
+
+class AutoDocking:
+    BATTERY_THRESHOLD = 4
+
+    def __init__(self):
+        self._ros_node = rospy.init_node('dock_drive_client_py', anonymous=True)
+        self._client = actionlib.SimpleActionClient('dock_drive_action', AutoDockingAction)
+        self._diagnostic_agg_sub = rospy.Subscriber("/diagnostics_agg", DiagnosticArray, self._listen_batteries)
+        self._go_docking = False
```

Comment:
For a first version this might be OK, but I would suggest to use a more descriptive name, and make sure the behavior is the intended one. With the current implementation the robot will go docking only once and never dock again.

For example, here this variable will be set to true when the docking action has started, so a more descriptive name would be `_docking_action_started` or `_docking_in_progress`. In that case, you would need to make sure that it is set to false again once the action finishes. But you would also need another flag that tells you if the robot is already docked and charging, so that you don't trigger the auto docking in the case it is charging (you could read the battery state for that).

So the logic will be something like:
If not charging and not docking_in_progress then dock (if the battery level is low).

Reply:
I forgot to change the state of `_go_docking` when is called `_doneCb`. Changing that state there I think the Leonardo is going to behave as expected.